modules/aws/node/main.tf

```diff
@@ -149,3 +159,38 @@ resource "aws_network_interface" "this" {
     "node.k8s.amazonaws.com/no_manage" = "true"
   }
 }
+
+resource "kubernetes_job" "wait" {
```

Comment:
I think something like this is quite a neat way of resolving the problem.  It makes sense for the `node` module to only be done when the node has actually joined the cluster.